pr_testing/test_multiple_prs.sh

```diff
@@ -79,19 +79,47 @@ function process_changed_files() {
   sort -u "$directlyChangedFiles" $WORKSPACE/indirectly-changed-files.txt > "$allChangedFiles"
 }
 
+function is_in_array() {
+    local value="$1"
+    shift
+    local array=("$@")
+
+    for item in "${array[@]}"; do
+        if [[ "$item" == "$value" ]]; then
+            return 0  # Found match
+        fi
+    done
+    return 1  # No match
+}
+
+SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"  # Absolute path to script
+CMS_BOT_DIR=$(dirname ${SCRIPTPATH})  # To get CMS_BOT dir path
+
+readarray -t ALL_GPU_TYPES < ${CMS_BOT_DIR}/gpu_flavors.txt
+
+declare -a ENABLE_GPU_FLAVORS
+for ex_type in ${EXTRA_RELVALS_TESTS} ; do
+  [ $(echo ${ENABLE_BOT_TESTS} | tr ',' ' ' | tr ' ' '\n' | grep "^${ex_type}$" | wc -l) -gt 0 ] || continue
+  ex_type_lc=$(echo $ex_type | tr '[A-Z]' '[a-z]')
+  if is_in_array "$ex_type_lc" "${ALL_GPU_TYPES[@]}" ; then
+    ENABLE_GPU_FLAVORS+=( $ex_type )
+    VAR_NAME="MATRIX_EXTRAS_${ex_type}"
+    if [ -z "${!VAR_NAME}" ]; then
+      eval "$VAR_NAME=${MATRIX_EXTRAS_GPU}"
+    fi
+  fi
+done
 # Constants
 echo LD_LIBRARY_PATH=${LD_LIBRARY_PATH} || true
 ls ${LD_LIBRARY_PATH} || true
-SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"  # Absolute path to script
-CMS_BOT_DIR=$(dirname ${SCRIPTPATH})  # To get CMS_BOT dir path
 export SCRAM_PREFIX_PATH=${CMS_BOT_DIR}/das-utils
 source ${CMS_BOT_DIR}/cmsrep.sh
 CACHED=${WORKSPACE}/CACHED            # Where cached PR metada etc are kept
 PR_TESTING_DIR=${CMS_BOT_DIR}/pr_testing
 COMMON=${CMS_BOT_DIR}/common
 CONFIG_MAP=$CMS_BOT_DIR/config.map
 [ "${USE_IB_TAG}" != "true" ] && export USE_IB_TAG=false
-[ "${EXTRA_RELVALS_TESTS}" = "" ] && EXTRA_RELVALS_TESTS="GPU THREADING HIGH_STATS NANO"
+[ "${EXTRA_RELVALS_TESTS}" = "" ] && EXTRA_RELVALS_TESTS="THREADING HIGH_STATS NANO $(echo ${ALL_GPU_TYPES[@]} | tr '[a-z]' '[A-Z]')"
 EXTRA_RELVALS_TESTS=$(echo ${EXTRA_RELVALS_TESTS} | tr ' ' '\n' | grep -v THREADING | tr '\n' ' ')
```

Comment:
Can we also drop `GPU` from `EXTRA_RELVALS_TESTS` here? and then we can avoid few ` [ $ex_type = "GPU" ] && continue` later?

Reply:
Keeping "GPU" makes triggering GPU unit tests easier, see https://github.com/cms-sw/cms-bot/blob/e5b63dd2c69c0512d735f2f30e78cd89c248fb5e/pr_testing/test_multiple_prs.sh#L1343-L1345

Reply:
Alternatively I can check if any of `${ALL_GPU_TYPES}` is in `${ENABLE_BOT_TESTS}`

Reply:
you already have this information in `ENABLE_GPU_FLAVORS`. If ENABLE_GPU_FLAVORS is set then gpu tests are requested

Reply:
@iarspider , no need to have GPU in EXTRA_RELVALS_TESTS